contracts/erc1155/ERC1155YungWkndBase.sol

```diff
@@ -0,0 +1,358 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity 0.8.10;
+
+import "./ERC1155Base.sol";
+import "../metadata/MetadataEncryption.sol";
+import "../tokenManager/interfaces/IPostTransfer.sol";
+import "../tokenManager/interfaces/IPostBurn.sol";
+import "./interfaces/IHighlightRenderer.sol";
+import "../utils/ERC1155/ERC1155URIStorageUpgradeable.sol";
+import "./interfaces/IERC1155YungWkndMint.sol";
+
+/**
+ * @title Generalized Base ERC1155
+ * @author highlight.xyz
+ * @notice Generalized Base NFT smart contract
+ */
+abstract contract ERC1155YungWkndBase is ERC1155Base, ERC1155URIStorageUpgradeable, IERC1155YungWkndMint {
+    using EnumerableSet for EnumerableSet.AddressSet;
+
+    uint256 _tokenCount;
```

Comment:
Guessing this tracks the number of unique tokens on the 1155 (distinct `tokenID`s), rather than the sum of quantities across all tokens? The minting logic should be updated to reflect that